network/transport/v2/senders.go

```diff
@@ -46,3 +47,84 @@ func (p *protocol) sendGossipMsg(id transport.PeerID, refs []hash.SHA256Hash) er
 		},
 	}})
 }
+
+func (p *protocol) sendTransactionListQuery(id transport.PeerID, refs []hash.SHA256Hash) error {
+	conn := p.connectionList.Get(grpc.ByConnected(), grpc.ByPeerID(id))
+	if conn == nil {
+		// shouldn't happen
+		return errors.New("no connection available")
+	}
+
+	// there shouldn't be more than a ~650 in there, this will fit in a single message
+	refsAsBytes := make([][]byte, len(refs))
+	for i, ref := range refs {
+		refsAsBytes[i] = ref.Slice()
+	}
+
+	envelop := &Envelope_TransactionListQuery{
+		TransactionListQuery: &TransactionListQuery{
+			Refs: refsAsBytes,
+		},
+	}
+
+	conversation := p.cMan.conversationFromEnvelope(envelop)
```

Comment:
must say `conversationFromEnvelope` is a bit of an oddball func (switching on types, setting fields). Maybe nicer to make the applicable messages implement an interface, so it can be solved a bit safer/generic?

```go
type Conversatable interface { // or whatever
   SetConversationID(cid conversationID)
}

// Provide this for every message that supports a conversation ID
func (envelope *Envelope_TransactionListQuery) SetConversationID(cid conversationID) {
	envelope.TransactionListQuery.ConversationID = cid.slice()
}
```

Reply:
refactored